src/main/scala/is/hail/expr/FunctionRegistry.scala

```diff
@@ -1164,6 +1164,32 @@ object FunctionRegistry {
     """,
     "p" -> "Probability")
 
+  register("rpois", { (lambda: Double) => rpois(lambda) },
+    """
+    Returns a random draw from a poisson distribution with parameter ``lambda``. This function is non-deterministic.
+    """,
+    "lambda" -> "Mean of poisson distribution. Must be non-negative.")
+  register("dpois", { (x: Int, lambda: Double) => dpois(x, lambda) },
+    """
+    Returns the :math:`P(X = x)` from a poisson distribution with parameter ``lambda``. This function is non-deterministic.
+    """,
+    "x" -> "Non-negative integer to compute the probability density.",
+    "lambda" -> "Mean of poisson distribution. Must be non-negative.")
+  register("ppois", { (x: Int, lambda: Double, lowerTail: Boolean) => ppois(x, lambda, lowerTail) },
+    """
+    Returns left-tail probability p for which p = :math:`P(X \leq x)` with :math:`X` a poisson random variable with parameter ``lambda``. This function is non-deterministic.
+    """,
+    "x" -> "Non-negative integer to compute the probability density at.",
+    "lambda" -> "Mean of poisson distribution. Must be non-negative.",
+    "lowerTail" -> "If false, returns the :math:`P(X > x)`")
+  register("qpois", { (p: Double, lambda: Double, lowerTail: Boolean) => qpois(p, lambda, lowerTail) },
+    """
+    Returns the largest integer x for which ``p`` = :math:`P(X \leq x)` with :math:`X` a poisson random variable with parameter ``lambda``. Inverse of ppois. This function is non-deterministic.
```

Comment:
This doesn't make sense as written: since `p` there may be no such integer achieving equality.

Also, be consistent on p vs ``p``, look at other functions.